File: ontrack-service/src/test/java/net/nemerosa/ontrack/service/support/property/TestProperty.java
package net.nemerosa.ontrack.service.support.property;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class TestProperty {

    private final String value;

    public static TestProperty of(String value) {
        return new TestProperty(value);
    }
}

File: ontrack-extension-api/src/testFixtures/kotlin/net/nemerosa/ontrack/extension/api/support/TestProperty.kt
package net.nemerosa.ontrack.extension.api.support

import net.nemerosa.ontrack.model.json.schema.JsonSchemaString
import net.nemerosa.ontrack.model.support.ConfigurationProperty

class TestProperty(
    @JsonSchemaString
    override val configuration: TestConfiguration,
    val value: String
) : ConfigurationProperty<TestConfiguration> {

    companion object {
        @JvmStatic
        fun of(value: String) = of(
            TestConfiguration.config("test"),
            value
        )

        @JvmStatic
        fun of(configuration: TestConfiguration, value: String) = TestProperty(
            configuration,
            value
        )
    }
}